pybuildkite/jobs.py

```diff
@@ -16,19 +16,18 @@ def __init__(self, client, base_url):
         self.client = client
         self.path = base_url + "/organizations/{}/pipelines/{}/builds/{}/jobs/{}/"
 
-    def get_job_log(self, organization, pipeline, build, job, get_text=False, get_html=False):
+    def get_job_log(self, organization, pipeline, build, job, log_format = 'text/html'):
```

Comment:
Does this need an enum?

Reply:
I see what you’re going for. I think we should go with the original one, then make a check for if get_text: set the correct header else if get_html: same thing. But have a check that if both are set to true we throw an exception. And then default to what we had originally. This was the user doesnt need to know the string and we just provide it in the method, thoughts?

Reply:
I'll change it to match what you want, but stylistically I feel it's the wrong choice.

```python
    def get_job_log(self, organization, pipeline, build, job, log_format = LogFormat.html):
```

means the user doesn't need to know the string, if we don't do validation we can use the validation already built into the API to give us the check (so if they ever add in new formats for whatever reason the client library still works, as a user can pass in a string if they really have to).

```python
    def get_job_log(self, organization, pipeline, build, job, get_text = False, get_html=False, get_new_format=False):
```

is pretty ugly and what's more the client is now missing functionality without a workaround until this is added in.

Also to a new person coming to the client what does this mean:

```python
    jobs.get_job_log("org_slug", "pipeline_name", "build_id", "job_number", true, true):
```
That I can get both text and html?

```python
    jobs.get_job_log("org_slug", "pipeline_name", "build_id", "job_number", false, false):
```

that I get an empty response?

with the top way these ambiguities aren't possible

Reply:
Great suggestion! I like your idea more, thanks for the explanation. Let's go with that- and I think it's nicer with the enum. Only open question for me now is how we handle importing the enum. The user will have to explicitly import it from this class unless we import it to `buildkite`. Thoughts?

Reply:
Off the top of my head there's a few choices

```python
buildkite.jobs().BuildState
```

Looks a little weird to me but it stays on the jobs class. Breaks the usual dynamic of importing though.

```python
buildkite.LogFormat
```

probably my preference, as it matches the BuildState example in the README.

`BuildState` does have the word `Build` in though which is the namespace it's under so maybe it should be 

```python
buildkite.JobLogFormat
```

Reply:
Okay we can do that, we'll have to import the Enum into the `buildkite` class though. But we do that for Build State so I think it's fine 👍